builtin/joined_dataset.cc

```diff
@@ -59,6 +59,30 @@ struct JoinedDataset::Itl
         //ML::compact_vector<RowHash, 2> rowHashes;   ///< Row hash from input datasets
     };
 
+     struct JoinedRowStream : public RowStream {
+
+        JoinedRowStream(JoinedDataset::Itl* source) : source(source)
+        {
+            
+        }
+
+        virtual std::shared_ptr<RowStream> clone() const{
+            auto ptr = std::make_shared<JoinedRowStream>(source);
+            return ptr;
+        }
+
+        virtual void initAt(size_t start){
+            iter = source->rows.begin() + start;
```

Comment:
I purposefully dont check for outofbounds for performance reasons. The existing interface for datasets means that we always have an exact Row Count, so the executor knows exactly how many times it should call next on the stream. I'll add a comment to specify that.